### ISSUE TITLE
Keras model not training, while tflearn model does.

### ISSUE THREAD

**Issue by Khushmeet** [fchollet/keras#7223]
I am running a 1D convolution on text, using the following model in keras

```
model = Sequential()
model.add(Embedding(input_dim=vocab_size, output_dim=256, input_length=100))
model.add(Conv1D(filters=128, kernel_size=3, padding='SAME', activation='relu', kernel_regularizer=keras.regularizers.l2(0.1)))
model.add(Conv1D(filters=128, kernel_size=7, padding='SAME', activation='relu', kernel_regularizer=keras.regularizers.l2(0.1)))
model.add(Conv1D(filters=128, kernel_size=7, padding='SAME', activation='relu', kernel_regularizer=keras.regularizers.l2(0.1)))
model.add(MaxPool1D(pool_size=3))
model.add(Dropout(0.5))
model.add(Flatten())
model.add(Dense(3, activation='softmax'))
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
```

My issue is that, the model is not training whatever I do. But the surprising part is that when i try to run the same model, with exact same configuration in tflearn, the models gets accuracy upto 95%. 

Why is that so?
Here are my training results.
<img width="988" alt="screen shot 2017-07-03 at 11 01 38 pm" src="https://user-images.githubusercontent.com/8711912/27802805-a142e5d6-6043-11e7-937e-fda27531d71b.png">

Running on macOS sierra, python 3.6.0, tensorflow - 1.2.1, keras - 2.0.4 (latest)

**Comment by taehoonlee** [comment]
Is not it a high-l2 problem?

**Comment by Khushmeet** [comment]
What should be appropriate value for l2?
Is 0.1  too high?
I tried with 0.01, but it gave me same results.

**Comment by taehoonlee** [comment]
There is no standard on how to choose hyper-parameters. But 0.01 looks still high to me.
